R/fit.R

```diff
@@ -42,6 +45,15 @@ CmdStanFit <- R6::R6Class(
     },
 
     metadata = function() {
+      if (!length(self$output_files(include_failed = FALSE))) {
+        if (self$runset$method() == "sample") {
+          stop("No chains finished successfully. Unable to retrieve the metadata.")
+        } else if (self$runset$method() == "generate_quantities") {
+          stop("Generating quantities for all MCMC chains failed. Unable to retrieve the metadata.", call. = FALSE)
+        } else {
+          stop("Fitting failed. Unable to retrieve the metadata.")
+        }
```

Comment:
What do you think about just simplifying this and just using `stop("Fitting failed. Unable to retrieve the metadata.")` as the error message for all methods instead of branching? (If you would rather keep the separate messages that's fine. )

Reply:
Simplifying sounds good.